sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -480,14 +509,24 @@ protected EventProcessorClient()
             Argument.AssertNotNull(partitionManager, nameof(partitionManager));
             Argument.AssertNotNull(connectionFactory, nameof(connectionFactory));
 
-            clientOptions = clientOptions?.Clone() ?? new EventProcessorClientOptions();
+            clientOptions = clientOptions ?? new EventProcessorClientOptions();
+
+            ProcessingConsumerOptions = new EventHubConsumerClientOptions
+            {
+                RetryOptions = clientOptions.RetryOptions
+            };
+
+            ProcessingReadEventOptions = new ReadEventOptions
+            {
+                MaximumWaitTime = clientOptions.MaximumWaitTime,
+                TrackLastEnqueuedEventProperties = clientOptions.TrackLastEnqueuedEventProperties
+            };
 
             ConnectionFactory = connectionFactory;
             FullyQualifiedNamespace = fullyQualifiedNamespace;
             EventHubName = eventHubName;
             ConsumerGroup = consumerGroup;
             StorageManager = partitionManager;
```

Comment:
Still using a `PartitionManager` in this `internal` constructor so we can use the `InMemoryPartitionManager` when testing the processor.